lib/Dialect/FIRRTL/Transforms/LowerIntrinsics.cpp

```diff
@@ -161,6 +169,32 @@ class CirctClockInverterConverter : public IntrinsicConverter {
   }
 };
 
+class CirctClockDividerConverter : public IntrinsicConverter {
+public:
+  using IntrinsicConverter::IntrinsicConverter;
+
+  bool check() override {
+    return hasNPorts(2) || namedPort(0, "in") || namedPort(1, "out") ||
+           hasNParam(1) || namedIntParam("POW_2");
+  }
+
+  LogicalResult convert(InstanceOp inst) override {
+    auto pow2 = cast<IntegerAttr>(
+        getNamedParam(mod.getParameters(), "POW_2").getValue());
+
+    ImplicitLocOpBuilder builder(inst.getLoc(), inst);
+    auto in = builder.create<WireOp>(inst.getResult(0).getType()).getResult();
+    inst.getResult(0).replaceAllUsesWith(in);
+    auto out = builder.create<ClockDividerIntrinsicOp>(in, pow2);
+    auto name = inst.getInstanceName();
+    Value outWire = builder.create<WireOp>(out.getType(), name).getResult();
```

Comment:
This extra wire only exists to carry the name given to the instance, is that right?

What sorts of names end up here, could they be handled by Chisel inserting a node with the desired name (or not, if naming it doesn't matter)?

Asking because:

I'm working to move to intrinsic expression/statements vs instance-of-intmodule -- where instead of lowering to a wire the user would generate something like:
```
node myname = intrinsic clock_divider.... 
```
(syntax details pending)

Does that generally seem like it works for the use case motivating these wires being inserted (here and clock inverter IIRC)?

Reply:
That would be great, what he have here are pretty much hacks to preserve a `sv.namehint` and push it into the op later.

Reply:
Great, thanks for the explanation/context and discussion offline 👍 .